django_project/species/models.py

```diff
@@ -57,6 +83,39 @@ class Meta:
         verbose_name_plural = "Taxa"
         db_table = "taxon"
 
+    def save(self, *args, **kwargs):
+        print(self.graph_icon)
+        if self.graph_icon:
+            graph_icon = self.graph_icon.read()
+            topper_icon = graph_icon.replace(
```

Comment:
Please update the code to use regex instead. Users should also be able to upload non-black svg files. Use this function :

```
def replace_fill_color(data, new_color):
    fill_pattern = re.compile(br'fill="[^"]*"', re.IGNORECASE)
    return fill_pattern.sub(br'fill="' + new_color.encode() + br'"', data)
```